lib/backend/k8s/resources/nodebgppeer_test.go

```diff
@@ -0,0 +1,131 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources_test
+
+import (
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/resources"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Node BGP conversion methods", func() {
+
+	converter := resources.NodeBGPPeerConverter{}
+
+	It("should convert an empty ListInterface", func() {
+		node, name, err := converter.ListInterfaceToNodeAndName(
+			model.NodeBGPPeerListOptions{},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal(""))
+		Expect(name).To(Equal(""))
+	})
+
+	It("should convert a List interface with a Node name only", func() {
+		node, name, err := converter.ListInterfaceToNodeAndName(
+			model.NodeBGPPeerListOptions{
+				Nodename: "node",
+			},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal("node"))
+		Expect(name).To(Equal(""))
+	})
+
+	It("should convert a List interface with a PeerIP only", func() {
+		node, name, err := converter.ListInterfaceToNodeAndName(
+			model.NodeBGPPeerListOptions{
+				PeerIP: net.MustParseIP("1.2.3.4"),
+			},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal(""))
+		Expect(name).To(Equal("1-2-3-4"))
+	})
+
+	It("should convert a List interface with node and PeerIP (IPv4)", func() {
+		node, name, err := converter.ListInterfaceToNodeAndName(
+			model.NodeBGPPeerListOptions{
+				Nodename: "nodeX",
+				PeerIP:   net.MustParseIP("1.2.3.40"),
+			},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal("nodeX"))
+		Expect(name).To(Equal("1-2-3-40"))
+	})
+
+	It("should convert a List interface with node and PeerIP (IPv6)", func() {
+		node, name, err := converter.ListInterfaceToNodeAndName(
+			model.NodeBGPPeerListOptions{
+				Nodename: "nodeX",
+				PeerIP:   net.MustParseIP("1::2:3:4"),
+			},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal("nodeX"))
+		Expect(name).To(Equal("1--2-3-4"))
+	})
+
+	It("should convert a Key with node and PeerIP (IPv4)", func() {
+		node, name, err := converter.KeyToNodeAndName(
+			model.NodeBGPPeerKey{
+				Nodename: "nodeY",
+				PeerIP:   net.MustParseIP("1.2.3.50"),
+			},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal("nodeY"))
+		Expect(name).To(Equal("1-2-3-50"))
+	})
+
+	It("should convert a Key with node and PeerIP (IPv6)", func() {
+		node, name, err := converter.KeyToNodeAndName(
+			model.NodeBGPPeerKey{
+				Nodename: "nodeY",
+				PeerIP:   net.MustParseIP("aa:ff::0012"),
+			},
+		)
+		Expect(err).To(BeNil())
+		Expect(node).To(Equal("nodeY"))
+		Expect(name).To(Equal("aa-ff--0012"))
+	})
+
+	It("should convert a valid node name and resource name to a Key (IPv4)", func() {
+		key, err := converter.NodeAndNameToKey("nodeA", "1-2-3-4")
+		Expect(err).To(BeNil())
+		Expect(key).To(Equal(model.NodeBGPPeerKey{
+			Nodename: "nodeS",
```

Comment:
Well that's just embarrassing - no test suite defined, so not actually running any of these tests.  Will fix (and fix up broken tests).